control-plane/api/v1alpha1/servicedefaults_types.go

```diff
@@ -183,6 +190,18 @@ type ServiceDefaultsDestination struct {
 	Port uint32 `json:"port,omitempty"`
 }
 
+// EnvoyExtension has configuration for an extension that patches Envoy resources.
+type EnvoyExtension struct {
+	Name     string `json:"name,omitempty"`
+	Required bool   `json:"required,omitempty"`
+	// +kubebuilder:validation:Type=object
+	// +kubebuilder:validation:Schemaless
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Arguments json.RawMessage `json:"arguments,omitempty"`
+}
+
+type EnvoyExtensions []EnvoyExtension
```

Comment:
I'll add a comment to this public type

Reply:
Since this type is shared between ServiceDefaults and ProxyDefaults we should probably move the type definitions to `api/v1alpha1/shared_types.go` and associated `validate` and `toConsul` methods. It is where we have defined types that are reused across CRDs.